core/history.go

```diff
@@ -1,128 +1,108 @@
 package core
 
 import (
-	"bytes"
-	"encoding/binary"
 	"errors"
 
 	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie"
 	"github.com/NethermindEth/juno/db"
-	"github.com/NethermindEth/juno/utils"
 )
 
-var ErrCheckHeadState = errors.New("check head state")
+var ErrHistoricalTrieNotSupported = errors.New("cannot support historical trie")
 
-type history struct {
-	txn db.IndexedBatch
+type stateHistory struct {
```

Comment:
Initially, it looks like you're removing state snapshot and use state history instead. But instead, I think you're actually removing state history, replace it with state snapshot implementation and rename to state history. Please next time, break this into 2 PRs:
- Remove state history, replace by the existing state snapshot
- Rename state snapshot to state history

We can actually achieve the same outcomes with much simpler diff.